build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/AndroidCompose.kt

```diff
@@ -59,9 +58,11 @@ internal fun Project.configureAndroidCompose(
 private fun Project.buildComposeMetricsParameters(): List<String> {
     val metricParameters = mutableListOf<String>()
     val enableMetricsProvider = project.providers.gradleProperty("enableComposeCompilerMetrics")
+    val projectPath = project.path.replace(':', '_').drop(1)
+
     val enableMetrics = (enableMetricsProvider.orNull == "true")
     if (enableMetrics) {
-        val metricsFolder = File(project.buildDir, "compose-metrics")
+        val metricsFolder = rootProject.buildDir.resolve("compose-metrics/$projectPath")
```

Comment:
`projectPath` will contain invalid characters on Windows: `:`.

Does this work?

```suggestion
        val metricsFolder = rootProject.buildDir.resolve(projectDir.relativeTo(rootDir))
```

Reply:
Ooops, I did not see the `project.path.replace(':', '_').drop(1)` at first.
Both solutions are valid here.
One potential issue (that does not appear to happen in this project) is that replacing path chars could also lead to collision, but is very unlikely. (e.g. `:foo:bar_baz` and `:foo_bar:baz`)

This would not happen with the relative directory method, but will add maybe cumbersome directories 🤷 
```kotlin
val relativePath = projectDir.relativeTo(rootDir)
val metricsFolder = rootProject.buildDir.resolve("compose-metrics").resolve(relativePath)
val reportsFolder = rootProject.buildDir.resolve("compose-reports").resolve(relativePath)
```

And the output:
![image](https://github.com/android/nowinandroid/assets/1921278/c1bc5a3e-a5df-4210-9c70-aa959d5621e8)

Reply:
I agree it's cumbersome, but since people might be copying the solution from this sample, let's make it safe, so it doesn't create confusion.

I think it's still beneficial to have it in one folder with subfolders than finding the metrics in each module.